app/src/main/java/org/schabi/newpipe/ui/components/items/ItemList.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.rememberNestedScrollInteropConnection
+import androidx.compose.ui.res.stringResource
+import androidx.fragment.app.FragmentActivity
+import androidx.preference.PreferenceManager
+import androidx.window.core.layout.WindowWidthSizeClass
+import my.nanihadesuka.compose.LazyColumnScrollbar
+import org.schabi.newpipe.R
+import org.schabi.newpipe.extractor.InfoItem
+import org.schabi.newpipe.extractor.playlist.PlaylistInfoItem
+import org.schabi.newpipe.extractor.stream.StreamInfoItem
+import org.schabi.newpipe.info_list.ItemViewMode
+import org.schabi.newpipe.ui.components.items.playlist.PlaylistListItem
+import org.schabi.newpipe.ui.components.items.stream.StreamListItem
+import org.schabi.newpipe.util.DependentPreferenceHelper
+import org.schabi.newpipe.util.NavigationHelper
+
+@Composable
+fun ItemList(
+    items: List<InfoItem>,
+    mode: ItemViewMode = determineItemViewMode(),
+    listHeader: LazyListScope.() -> Unit = {}
+) {
+    val context = LocalContext.current
+    val onClick = remember {
+        { item: InfoItem ->
+            val fragmentManager = (context as FragmentActivity).supportFragmentManager
+            if (item is StreamInfoItem) {
+                NavigationHelper.openVideoDetailFragment(
+                    context, fragmentManager, item.serviceId, item.url, item.name, null, false
+                )
+            } else if (item is PlaylistInfoItem) {
+                NavigationHelper.openPlaylistFragment(
+                    fragmentManager, item.serviceId, item.url, item.name
+                )
+            }
+        }
+    }
+
+    // Handle long clicks for stream items
+    // TODO: Adjust the menu display depending on where it was triggered
+    var selectedStream by remember { mutableStateOf<StreamInfoItem?>(null) }
+    val onLongClick = remember {
+        { stream: StreamInfoItem ->
+            selectedStream = stream
+        }
+    }
+    val onDismissPopup = remember {
+        {
+            selectedStream = null
+        }
+    }
+
+    val showProgress = DependentPreferenceHelper.getPositionsInListsEnabled(context)
+    val nestedScrollModifier = Modifier.nestedScroll(rememberNestedScrollInteropConnection())
+
+    if (mode == ItemViewMode.GRID) {
+        // TODO: Implement grid layout using LazyVerticalGrid and LazyVerticalGridScrollbar.
+    } else {
+        val state = rememberLazyListState()
+
+        LazyColumnScrollbar(state = state) {
+            LazyColumn(modifier = nestedScrollModifier, state = state) {
+                listHeader()
+
+                items(items.size) {
+                    val item = items[it]
+
+                    if (item is StreamInfoItem) {
+                        val isSelected = selectedStream == item
```

Comment:
This should be `selectedStream === item`, right? Otherwise it's going to compare the fields, which might be slower (and is also not what we want).